QuestionBot/QuestionBot.UnitTests/Model/InMemoryStoreTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+using System.Collections.Generic;
+using QuestionBot.Model;
+
+namespace QuestionBot.UnitTests.Model {
+
+    [TestFixture]
+    public class InMemoryStoreTests{
+        private IStore _storeTest;
+        private IEnumerable<IRecord> _retrievedRecords = new List<IRecord>();
+
+        [SetUp]
+        public void Setup(){
+            _storeTest = new InMemoryStore();
+        }
+
+        [Test]
+        public void New_question_creates_record(){
+            string question = "What is 1+2?";
+
+            IRecord questionRecord = _storeTest.CreateRecord(question);
+
+            Assert.AreEqual(questionRecord.Question, question);
+            Assert.AreNotEqual(questionRecord.TimeAsked, DateTime.MinValue);
+            Assert.AreNotEqual(questionRecord.ID,0);
+        }
+
+        [Test]
+        public void Updating_question_with_answer_stores_my_answer(){
+            string question = "What is 1+2?";
+            string answer = "3";
+
+            IRecord questionRecord = _storeTest.CreateRecord(question);
+            IRecord answeredRecord = _storeTest.UpdateRecord(questionRecord.ID, answer);
+
+            Assert.AreEqual(answeredRecord.Question, question);
+            Assert.AreEqual(answeredRecord.Answer, answer);
+            Assert.AreNotEqual(answeredRecord.TimeAsked, DateTime.MinValue);
+            Assert.AreNotEqual(answeredRecord.TimeAnswered, DateTime.MinValue);
+            Assert.AreNotEqual(answeredRecord.ID,0);
+        }
+
+        [Test]
+        public void Updating_question_with_answer_updates_the_record(){
+            string question = "What is 2+3?";
+            string answer = "5";
+
+            IRecord questionRecord = _storeTest.CreateRecord(question);
+            IRecord answeredRecord = _storeTest.UpdateRecord(questionRecord.ID, answer);
+
+            Assert.AreEqual(answeredRecord.Question, questionRecord.Question);
+            Assert.AreEqual(answeredRecord.Answer, questionRecord.Answer);
+            Assert.AreEqual(answeredRecord.ID, questionRecord.ID);
+            Assert.AreEqual(answeredRecord.TimeAsked, questionRecord.TimeAsked);
+            Assert.AreEqual(answeredRecord.TimeAnswered, questionRecord.TimeAnswered);
```

Comment:
Optional: You could clean up this test and the `GetRecords_returns_all_question_records` test by adding a small helper that takes 2 IRecords and compares them for equality.